src/workspaces/LockWorkspaceModal/LockWorkspaceModal.tsx

```diff
@@ -23,21 +24,17 @@ const LockWorkspaceModal = (props: LockWorkspaceModalProps) => {
   const [togglingLock, setTogglingLock] = useState(false);
   const helpText = isLocked ? 'Unlock Workspace' : 'Lock Workspace';
 
-  const onFailureDismiss = () => {
-    setTogglingLock(false);
-    onDismiss();
-  };
-
-  const toggleWorkspaceLock = withErrorReportingInModal(
-    'Error toggling workspace lock',
-    onFailureDismiss
-  )(async () => {
-    setTogglingLock(true);
-    isLocked
-      ? await Ajax().Workspaces.workspace(namespace, name).unlock()
-      : await Ajax().Workspaces.workspace(namespace, name).lock();
-    onDismiss();
-    onSuccess();
+  const toggleWorkspaceLock = withBusyState(setTogglingLock)(async () => {
+    try {
+      isLocked
+        ? await Ajax().Workspaces.workspace(namespace, name).unlock()
+        : await Ajax().Workspaces.workspace(namespace, name).lock();
+      onDismiss();
+      onSuccess();
+    } catch (error) {
+      reportError('Error toggling workspace lock', error);
+      onDismiss();
+    }
```

Comment:
I refactored this section with help from Blake because the previous implementation made it hard to mock out all the sources of error notifications for tests. This still has the same functionality and uses the existing `withBusyState` instead of manually setting `togglingLock` at the beginning and end of the "busy" period.